### DISCUSSION TITLE
Replace exisiting profile with packaged profile during install

### DISCUSSION THREAD

**Discussion by salimv** [elgatosf/elgato#35] [Help]
I include several profiles as part of the plugin package and declare those in the `profiles` property of `manifest.json`. When someone uninstalls the plugin thru the store, those profiles are not removed. Preference would be to remove it, but if it does not, that is not a big issue. But when the plugin is re-installed, I see that a `Copy` version of each of these plugins is created. Is there a way to replace existing profiles with the ones from the install package?

**Comment by GeekyEggo** [comment]
Hey @salimv, at the moment the Stream Deck API doesn't provide an interface for replacing profiles; this is intentional to ensure that any changes made to the profile, by the user, aren't lost. Understandably, for pre-defined profiles that are read-only, this limitation does not make sense and is potentially something we could look into in the future.

**Comment by salimv** [comment]
Thank you for the response. Is there any suggested workaround to delete readonly pre-installed profiles thru a script during install? I am new and not sure if there is any such hook offered. At the moment, the users of our plugin have the responsibility to delete pre-installed readonly profiles before re-installing the plugin to not create duplicates, and we are looking for ways for them to not have to do this. A workaround until a cleaner approach is develop is acceptable, but we are not sure if there is any workaround possible.
